src/client/components/UserEditor.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import { User } from '../../shared/User';
+import { remult } from '../utils/api-facade';
+import { Box } from '@mui/material';
+import { ErrorInfo } from 'remult';
+
+const userRepo = remult.repo(User);
+
+interface IProps {
+    user: User;
+    onClose: () => void;
+    onSaved: (user: User) => void;
+    create?: boolean;
+}
+export const UserEditor: React.FC<IProps> = ({ user, onClose, onSaved, create }) => {
+    const [data, setData] = React.useState<Partial<User>>(user);
+    const [errors, setErrors] = React.useState<ErrorInfo<User>>(null);
+    const handleClose = () => {
+        onClose();
+    };
+    const handleSave = async () => {
+        setErrors(null);
+
+        try {
+            const newUser = await userRepo.save(data, create);
+            onSaved(newUser);
+            handleClose();
+        }
+        catch (err: any) {
+
+            setErrors(err);
+        }
+    }
+    return (
+        <Dialog open={Boolean(user)} onClose={handleClose}>
+            <DialogTitle>{create ? "Create " : "Update "} User</DialogTitle>
+            <DialogContent>
+                <Box
+                    component="form"
+                    sx={{
+                        '& .MuiTextField-root': { m: 1 },
+                    }}
+                    noValidate
+                    autoComplete="off"
+                >
+                    <TextField
+                        autoFocus
+                        id="firstName"
```

Comment:
`[id]` is something to avoid. I bet you know that it creates implicitly a global variable named as the value of the attribute - among other issues with using an id in the DOM.

Reply:
It is better to use the name attribute.

Reply:
OK - I just took that from the MUI example - 
https://mui.com/components/text-fields/#basic-textfield

I didn't know and will not do that anymore